fem/gslib.cpp

```diff
@@ -0,0 +1,220 @@
+﻿// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#include "gslib.hpp"
+
+#ifdef MFEM_USE_GSLIB
+
+// Ignore warnings from the gslib header (GCC version)
+#ifdef MFEM_HAVE_GCC_PRAGMA_DIAGNOSTIC
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#endif
+
+#include "gslib.h"
+
+#ifdef MFEM_HAVE_GCC_PRAGMA_DIAGNOSTIC
+#pragma GCC diagnostic pop // remove for older GCC versions, e.g. 5.4.0
```

Comment:
According to GCC documentation, this pragma was introduced in GCC v4.6: see https://gcc.gnu.org/gcc-4.6/changes.html (search for `diagnostic push`). The macro `MFEM_HAVE_GCC_PRAGMA_DIAGNOSTIC` is actually defined for GCC >= v4.7: https://github.com/mfem/mfem/blob/5c03fad898b6f39037e3d8ec27db6a03c8d5954f/config/config.hpp#L28-L30

I'm not sure why 5.4.0 is given as an example here -- it should not be necessary.

Reply:
@jakubcerveny reported that `#pragma GCC diagnostic pop` specifically does not work on his machine

Reply:
@jakubcerveny -- can you comment on this?

Reply:
The problem on 5.4.0 is that once you "pop" the diagnostic flags, the warning "unused function" is no longer suppressed, even though the include was processed with the right flags. This was apparently a bug in gcc, because the unused functions were detected at a later stage in the compiler. We can probably ignore this because 5.4.0 is really old. I think we can just remove the comment as it causes confusion.